src/chainstate/stacks/index/marf.rs

```diff
@@ -481,19 +489,39 @@ impl MARF {
 
         let path = TriePath::from_key(key);
 
-        let leaf_opt_res = MARF::get_path(&mut self.storage, block_hash, &path)?;
+        let result = MARF::get_path(&mut self.storage, block_hash, &path);
 
         // restore
+        // Q: should this return on error? or return a corruption error?
```

Comment:
It could fail due to an I/O error.  I'm in favor of retaining the `?` here, and having whatever calls into the MARF do an `expect()` or `panic!()` if it fails.

Reply:
Sounds good